### PR TITLE
Adds additional validation to the UserProfile model.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Validates:
- The `alternate_user_id` is a valid ID (e.g., the user ID exists in our DB).
- The `alternate_user_id` is not the ID of the user.
- The `email` is not the email of the user.

### CODE REVIEW

File: dashboard/app/models/user_profile.rb [hunk 1]
```diff
@@ -58,11 +60,15 @@ def get_other_user_ids
...
-  # TODO(asher): Determine whether we should validate the existence of an
-  #   account with user_id of alternate_user_id.
   def add_other_user_id(alternate_user_id)
     existing_other_user_ids = get_other_user_ids
+
+    # Exit early if the ID does not make sense: the ID is already an alternate,
+    # the ID is invalid, or the ID is the user ID for this user.
     return if existing_other_user_ids.include? alternate_user_id
+    return unless User.find_by_id(alternate_user_id)
+    return if alternate_user_id == self.user.id
```
Comment (aoby) [inline dashboard/app/models/user_profile.rb:71]: Do we want these to fail silently, or should we raise an error?

Comment (aoby) [reply dashboard/app/models/user_profile.rb:71]: Since these rely on `user`, it's probably worth adding a `validates_presence_of :user`

Comment (ashercodeorg) [reply dashboard/app/models/user_profile.rb:71]: Given current usage, failing silently is the right action.

And +1 to validating the presence of user.

**Comment by ashercodeorg** [comment]
Adding validating presence of user. After CircleCI passes, I'll merge.

**Comment by ashercodeorg** [comment]
CircleCI failure (ChromeLatestWin7_bigGameVersions) unrelated.

**Pull request merged**